src/App.tsx

```diff
@@ -1,223 +1,97 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useCallback,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Errors } from './components/Errors';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterMethod, setFilterMethod] = useState('all');
+
+  const getTodosFromsServer = useCallback(async () => {
+    try {
+      if (user) {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      }
```

Comment:
to avoid nesting you also can write smth like this
```suggestion
    if (!user) return;
    
    try {
      const todosFromServer = await getTodos(user.id);

      setTodos(todosFromServer);
```